app/src/main/java/com/brainwallet/ui/screens/home/receive/ReceiveDialog.kt

```diff
@@ -0,0 +1,436 @@
+    val context = LocalContext.current
+    val wheelPickerFiatCurrencyState = rememberWheelPickerState(0)
+
+    LaunchedEffect(Unit) {
+        viewModel.onEvent(ReceiveDialogEvent.OnLoad(context))
+        viewModel.uiEffect.collect { effect ->
+            when (effect) {
+                is UiEffect.ShowMessage -> Toast.makeText(
+                    context,
+                    effect.message,
+                    Toast.LENGTH_SHORT
+                ).show()
+
+                else -> Unit
+            }
+        }
+    }
+
+    LaunchedEffect(Unit) {
+        delay(500)
+        wheelPickerFiatCurrencyState.scrollToIndex(state.getSelectedFiatCurrencyIndex())
+    }
+
+    LaunchedEffect(wheelPickerFiatCurrencyState) {
+        snapshotFlow { wheelPickerFiatCurrencyState.currentIndex }
+            .filter { it > -1 }
+            .distinctUntilChanged()
+            .debounce(700)
+            .collect {
+                Timber.i("wheelPickerFiatCurrencyState: currentIndex $it")
+
+                viewModel.onEvent(ReceiveDialogEvent.OnFiatCurrencyChange(state.fiatCurrencies[it]))
+            }
+    }
+
+    Column(
+        modifier = Modifier
+            .verticalScroll(rememberScrollState())
+            .fillMaxSize(),
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        CenterAlignedTopAppBar(
+            colors = TopAppBarDefaults.centerAlignedTopAppBarColors(
+                containerColor = BrainwalletTheme.colors.content //invert surface
+            ),
+            expandedHeight = 56.dp,
+            title = {
+                Text(
+                    text = stringResource(R.string.bottom_nav_item_buy_receive_title).uppercase(),
+                    style = BrainwalletTheme.typography.titleSmall
+                )
+            },
+            navigationIcon = {
+                if (state.moonpayWidgetVisible()) {
+                    IconButton(onClick = {
+                        viewModel.onEvent(ReceiveDialogEvent.OnSignedUrlClear)
+                    }) {
+                        Icon(
+                            Icons.AutoMirrored.Filled.ArrowBack,
+                            contentDescription = stringResource(R.string.back)
+                        )
+                    }
+                }
+            },
+            actions = {
+                IconButton(onClick = onDismissRequest) {
+                    Icon(
+                        Icons.Default.Close,
+                        contentDescription = stringResource(R.string.AccessibilityLabels_close),
+                        tint = BrainwalletTheme.colors.surface
+                    )
+                }
+            }
+        )
+
+        //moonpay widget
+        AnimatedVisibility(visible = state.moonpayWidgetVisible()) {
+            state.moonpayBuySignedUrl?.let { signedUrl ->
+                MoonpayBuyWidget(
+                    modifier = Modifier.height(500.dp),
```

Comment:
some how can't make it `.fillMaxHeight()` that's why hardcoded 500dp & also tested on small resolution

cc @kcw-grunt @josikie

Reply:
> some how can't make it `.fillMaxHeight()` that's why hardcoded 500dp & also tested on small resolution
> 
> cc @kcw-grunt @josikie

Oh. it thats the case @andhikayuana  , lets just have the Webview the way it was after we get the signed url (full screen with a close (X)) Its not the vision, but we need to get this feature out. We can talk to MoonPay about ways to embed.

Reply:
noted and also will revisit on this later